libcxx/include/fstream

```diff
@@ -337,6 +377,23 @@ private:
 
     return this;
   }
+
+  _LIBCPP_HIDE_FROM_ABI void __try_set_unbuffered_mode() {
```

Comment:
I think it would make sense to inline this function into `__do_open`. I would find it easier to understand the code since it's just a few lines and it's not really a function we could potentially reuse in other places (unlike the version below with arguments).